openff/toolkit/topology/molecule.py

```diff
@@ -6162,9 +6164,4 @@ def _generate_unique_atom_names(obj: Union[FrozenMolecule, HierarchyElement]):
     for atom in obj.atoms:
         symbol = atom.symbol
         element_counts[symbol] += 1
-        # TODO: It may be worth exposing this as a user option, i.e. to avoid multiple ligands
-        # parameterized with OpenFF clashing because they have atom names like O1x, H3x, etc.
-        # i.e. an optional argument could enable a user to `generate_unique_atom_names(blah="y")
-        # to have one ligand be O1y, etc.
-        # https://github.com/openforcefield/openff-toolkit/pull/1096#pullrequestreview-767227391
-        atom.name = symbol + str(element_counts[symbol]) + "x"
+        atom.name = symbol + str(element_counts[symbol]) + suffix
```

Comment:
It looks like there are two defined `generate_unique_atom_names` functions? (Messy merge?) This is unhappy because one of them is trying to stringify `None`

```python
>>> '' + None
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate str (not "NoneType") to str

Reply:
I'm somewhat confident (~70%) there should be two definitions of `generate_unique_atom_names` (one on FrozenMolecule and another on HierarchyElement). So I don't think the root cause here is a merge conflict, rather it's a genuine code issue (TM) caused by me merging in some code suggestions from the previous PR. 

I'm contacting @PEFrankel to see if he wants to carry this over the finish line. Otherwise I'll clean up my mess here :-P

Reply:
Okay yeah that would make sense - I may have glossed over that in the diff

Reply:
Still needs to be a way to prevent this concatenation, `+ suffix if suffix else ''` is probably lazy enough to work

Reply:
Yes, I've been using "else" flags in the calls since I planned on adding tests there, but this works as well.